docs/symboltable_serialization.md

```diff
@@ -0,0 +1,63 @@
+# Abstract
+In this document we are going to present how Jariko allows managing programs memory persistence using as approach the serialization.  
+As everyone knows the symbol table is an object containing the program memory, and generally speaking
+it is nothing more than a `Map<String, Value>` with String: the variable name, and Value: the value of it.  
+Symbol table has scope program and lives within the program, but when the program dies, if that program runs into an 
+activation group, we need to recover its state, for this reason we have thought about serialization.
+
+# Serialization model
+The following chart shows the objects involved with particular relevance to what happens over the time.  
+We suppose that Jariko will be run three different programs in the same thread.  
+The almost hidden objects (gray) are *not active*, the others are *active*, but all objects *are living*
+in the same instance of Jariko.
+
+![PD000236-SerializationModel_1.png](images/PD000236-SerializationModel_1.png)
+
+## InternalInterpreter
+For each interpreted PGM we have an instance of InternalInterpreter having its own instance of SymbolTable.  
+You can think about the follow calls stack: PGM1 -> PGM2 -> PGMi and as you can guess, PGMi is the current execution program.  
+
+## MemorySlice
+This object tells us that the SymbolTable and its own persistence are different things. The MemorySlice is nothing 
+of more complex than an association between the SymbolTable instance and the MemorySliceId.  
+MemorySliceId is identified just by following properties:
+ * Program
+ * Activation group  
+
+In conclusion, we can say that we will have an instance of MemorySlice for each SymbolTable.
+
+## MemorySliceMgr
+This object, behind the scenes, handles serialization and de-serialization mechanisms.
+
+### Deserialization
+For each SymbolTable belonging to a program running within an activation group, the initialization phase includes:
+ * The creation of MemorySlice in which we couple SymbolTable with MemorySliceId
+ * The reading from IMemorySliceStorage and retrieving of `Map<String, Value>` containing stored values associated to MemorySliceId
+ * The SymbolTable initialization using `Map<String, Value>` returned by the previous step
+
+### Serialization
+The serialization is triggered only at the end of the main program execution.  
+For each MemorySlice having property `persist=true` (program running into activation group), MemorySlice will be passed to
```

Comment:
not "program running into activation group" but "program finishing with RT indicator"

Reply:
ok